### ISSUE TITLE
Github Action docker compose build failed cause platform

### ISSUE THREAD

**Issue by minemos** [docker/compose#9877]
**Description**

docker compose build failed cause platform
I tried set `platform: linux/amd64` on docker-compose.yml and `DOCKER_DEFAULT_PLATFORM` to environment

Here is my actions
```yml
name: Deploy to ECS
on:
  push:
    branches:
      - main

jobs:
  deploy:
    runs-on: ubuntu-latest

    env:
      AWS_REGION: *
      ECS_CLUSTER_NAME: *
      ECR_REPOSITORY_NAME: *
      ECS_CONTAINER_NAME: *
      ECS_SERVICE_NAME: *

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Configure AWS credentials
        uses: aws-actions/configure-aws-credentials@v1
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws-region: ${{ env.AWS_REGION }}

      - name: Login to Amazon ECR
        id: login-ecr
        uses: aws-actions/amazon-ecr-login@v1

      - name: Build, tag, and push image to Amazon ECR
        id: build-image
        env:
          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
          IMAGE_TAG: latest
          ENV: ${{ secrets.ENV }}
        run: |
          echo "$ENV" > .env && chmod 777 .env
          docker version
          docker compose version
          docker info
          docker compose build
          docker tag kpop-backend-strapi:latest $ECR_REGISTRY/$ECR_REPOSITORY_NAME:$IMAGE_TAG
          docker push $ECR_REGISTRY/$ECR_REPOSITORY_NAME:$IMAGE_TAG
          echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY_NAME:$IMAGE_TAG"
      - name: Fill in the new image ID in the Amazon ECS task definition
        id: task-def
        uses: aws-actions/amazon-ecs-render-task-definition@v1
        with:
          task-definition: task-definition.json
          container-name: ${{ env.ECS_CONTAINER_NAME }}
          image: ${{ steps.build-image.outputs.image }}

      - name: Deploy Amazon ECS task definition
        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
        with:
          task-definition: ${{ steps.task-def.outputs.task-definition }}
          service: ${{ env.ECS_SERVICE_NAME }}
          cluster: ${{ env.ECS_CLUSTER_NAME }}
          wait-for-service-stability: true
```

And docker-compose.yml
```yml
version: '2.4'
services:
  strapi:
    build: .
    platform: linux/amd64
    ports:
      - "80:1337"
```

Dockerfile
```
FROM node:16-alpine
WORKDIR /app
COPY ./package.json ./
RUN yarn install
COPY . .
ENV NODE_ENV production
RUN yarn build
EXPOSE 1337
CMD ["yarn", "start"]
```

**Steps to reproduce the issue:**

1. Run actions to deploy

**Describe the results you received:**

```bash
Client:
 Version:           20.10.18+azure-1
 API version:       1.41
 Go version:        go1.18.6
 Git commit:        b40c2f6b5deeb11ac6c485c940865ee40664f0f0
 Built:             Thu Sep  8 08:19:02 UTC 20[22](https://github.com/kpop4planet/kpop-backend/actions/runs/3130813166/jobs/5081476380#step:5:23)
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.18+azure-1
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.6
  Git commit:       e4[23](https://github.com/kpop4planet/kpop-backend/actions/runs/3130813166/jobs/5081476380#step:5:24)27a6d3c55ceda3bd5475be7aae6036d02db3
  Built:            Thu Sep  8 22:50:10 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.13+azure-1
  GitCommit:        a17ec496a95e55601607ca50828147e8ccaeebf1
 runc:
  Version:          1.1.4
  GitCommit:        5fd4c4d144137e991c4acebb2146ab1483a979[25](https://github.com/kpop4planet/kpop-backend/actions/runs/3130813166/jobs/5081476380#step:5:26)
 docker-init:
  Version:          0.19.0
  GitCommit:        
service.platform should be part of the service.build.platforms: "linux/amd64"
```

**Describe the results you expected:**

Build success and deploy to ecs

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker compose version`:**

```
Docker Compose version 2.11.0+azure-1
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., 0.9.1+azure-1)
  compose: Docker Compose (Docker Inc., 2.11.0+azure-1)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 19
 Server Version: 20.10.18+azure-1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: a17ec496a95e55[60](https://github.com/kpop4planet/kpop-backend/actions/runs/3130826834/jobs/5081506698#step:5:61)1607ca50828147e8ccaeebf1
 runc version: 5fd4c4d144137e991c4acebb2146ab1483a97925
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.15.0-1020-azure
 Operating System: Ubuntu 20.04.5 LTS
 OSType: linux
 Architecture: x86_[64](https://github.com/kpop4planet/kpop-backend/actions/runs/3130826834/jobs/5081506698#step:5:65)
 CPUs: 2
 Total Memory: 6.781GiB
 Name: fv-az1[77](https://github.com/kpop4planet/kpop-backend/actions/runs/3130826834/jobs/5081506698#step:5:78)-[91](https://github.com/kpop4planet/kpop-backend/actions/runs/3130826834/jobs/5081506698#step:5:92)2
 ID: P7JB:LCAI:RKRH:MI44:TPU7:5IJN:OHVM:ANHL:TYPP:WNLY:WCRC:ASGL
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: githubactions
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details:**

It's running on github actions

**Comment by murakoshis** [comment]
```
Current runner version: '2.296.2'
Operating System
  Ubuntu
  20.04.5
  LTS
Runner Image
Runner Image Provisioner
GITHUB_TOKEN Permissions
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v2' (SHA:7884fcad6b5d53d10323aee724dc68d8b9096a2e)
Download action repository 'actions/setup-go@v2' (SHA:bfdd3570ce990073878bf10f6b2d79082de49492)
Download action repository 'actions/cache@v2' (SHA:937d24475381cd9c75ae6db12cb4e79714b926ed)
Download action repository 'golangci/golangci-lint-action@v2' (SHA:5c56cd6c9dc07901af25baab6f2b0d9f3b7c3018)
Download action repository 'aws-actions/configure-aws-credentials@0dbbed9302c857b576dd120df211eb34e72b7001' (SHA:0dbbed9302c857b576dd120df211eb34e72b7001)
Download action repository 'aws-actions/amazon-ecr-login@v1' (SHA:9149ade017c57f86dea2f76a01f8b2d5bd06b10f)
Download action repository 'yokawasa/action-setup-kube-tools@v0.8.2' (SHA:f7f73d7965074b0f5491b7210ff51e3254d42058)
```

Ubuntuバージョンは昨日と今日で特に変更なし

**Comment by milas** [comment]
Fixed in v2.11.1+. Please upgrade to latest and try again.
